moduleframework/mtf_scheduler.py

```diff
@@ -239,8 +239,8 @@ def avocado_general(self):
         avocado_args = ""  # when needed => specify HERE your additional stuff
         avocadoAction = "avocado {ACTION} {AVOCADO_ARGS}".format(
             ACTION=self.args.action, AVOCADO_ARGS=avocado_args)
-        bash = process.run("{AVOCADO} {a} {b}".format(
-            AVOCADO=avocadoAction, a=self.additionalAvocadoArg, b=" ".join(self.tests)), shell=True, ignore_status=True)
+        bash = subprocess.check_call("{AVOCADO} {a} {b}".format(
```

Comment:
Use try and expect CalledProcessError.https://docs.python.org/2/library/subprocess.html#subprocess.check_call